lib/scholar/decomposition/pca.ex

```diff
@@ -0,0 +1,252 @@
+    :components,
+    :explained_variance,
+    :explained_variance_ratio,
+    :singular_values,
+    :mean,
+    :num_components,
+    :num_features,
+    :num_samples
+  ]
+
+  fit_opts_schema = [
+    num_components: [
+      type: {:or, [:pos_integer, {:in, [nil]}]},
+      default: nil,
+      doc: ~S"""
+      Number of components to keep. If `:num_components` is not set, all components are kept:
+
+      $num\\_components = min(num\\_samples, num\\_features)$
+      """
+    ]
+  ]
+
+  transform_opts_schema = [
+    whiten: [
+      type: :boolean,
+      default: false,
+      doc: """
+      When true the result is multiplied by the square root of `:num_samples` and then
+      divided by the `:singular_values` to ensure uncorrelated outputs with unit component-wise variances.
+
+      Whitening will remove some information from the transformed signal (the relative variance scales of the components)
+      but can sometime improve the predictive accuracy of the downstream estimators by making their data respect some hard-wired assumptions.
+
+      """
+    ]
+  ]
+
+  fit_transform_opts_schema = fit_opts_schema ++ transform_opts_schema
+
+  @fit_opts_schema NimbleOptions.new!(fit_opts_schema)
+  @transform_opts_schema NimbleOptions.new!(transform_opts_schema)
+  @fit_transform_opts_schema NimbleOptions.new!(fit_transform_opts_schema)
+
+  @doc """
+  Fits a PCA for sample inputs `x`.
+
+  ## Options
+
+  #{NimbleOptions.docs(@fit_opts_schema)}
+
+  ## Returns
+
+  The function returns a struct with the following parameters:
+
+    * `:components` - Principal axes in feature space, representing the directions of maximum variance in the data.
+      Equivalently, the right singular vectors of the centered input data, parallel to its eigenvectors.
+      The components are sorted by `:explained_variance`.
+
+    * `:explained_variance` - The amount of variance explained by each of the selected components.
+      The variance estimation uses `:num_samples - 1` degrees of freedom.
+      Equal to `:num_components` largest eigenvalues of the covariance matrix of `x`.
+
+    * `:explained_variance_ratio` - Percentage of variance explained by each of the selected components.
+      If `:num_components` is not set then all components are stored and the sum of the ratios is equal to 1.0.
+
+    * `:singular_values` - The singular values corresponding to each of the selected components.
+      The singular values are equal to the 2-norms of the `:num_components` variables in the lower-dimensional space.
+
+    * `:mean` - Per-feature empirical mean, estimated from the training set.
+
+    * `:num_components` - It equals the parameter `:num_components`, or the lesser
+      value of `:num_features` and `:num_samples` if the parameter `:num_components` is `nil`.
+
+    * `:num_features` - Number of features in the training data.
+
+    * `:num_samples` - Number of samples in the training data.
+  """
+  deftransform fit(x, opts \\ []) do
```

Comment:
Did we settle on `train`?

Reply:
For this particular one, the `fit` fits better than the `train` in my opinion. However, if we want to stick to one name then I'm ok with the `train`.

Reply:
I agree, though I think unified naming is best for the end user. I would also be fine with `fit` everywhere. @josevalim I believe you advocated for `train`, thoughts on this case?

Reply:
To me the semantic difference is that `train` implies a more iterative process, while `fit` is more like a fixed calculation, so this differs on algorithm basis. However, for the end user the result is the same.

Reply:
I am fine either way, as long as we keep one. Perhaps fit is more general, so we should stabilize on that?

Reply:
Sounds good to me!

Reply:
So let's merge this as `fit` whenever ready and then rename the rest to `fit` in a separate PR?